### PR TITLE
VACMS-3323: Upgrade simplesamlphp library and Drupal module

### PR DESCRIPTION
**Pull request by ElijahLynn**
See #3323

This PR deletes over 4,000 files that were duplicated from vendor and were copied up outside of docroot. Now it just symlinks the docroot/vendor/simplesamlphp/{simplesamlphp,saml2} directories and copies the config/metadata folders so as not to have to duplicate them. 

This means it is not reviewable here and you need to check it out locally to review.

## Testing done
- [x] Deploy to DEV
- [x] Tested working login on DEV with my PIV card
- [x] Test that password change still works

### CODE REVIEW

**Comment by ElijahLynn** [comment]
Oh, there is a folder in root for 1.17.2 that I need to figure out what has changed as well and put 1.18.8 there instead.

**Review by swirtSJW — APPROVED** [review]
Changes all look good.   If it tests out on dev or staging, it will be good to merge.

**Comment by ElijahLynn** [comment]
Yay for tests...
Failure on features/user.feature:4 with:

> And I should see "Login with PIV or other Smartcard." in the "#edit-simplesamlphp-auth-login-link" element # Drupal\DrupalExtension\Context\MinkContext::assertElementContainsText()
> 
>       Element matching css "#edit-simplesamlphp-auth-login-link" not found. (Behat\Mink\Exception\ElementNotFoundException)

**Comment by ElijahLynn** [comment]
> Oh, there is a folder in root for 1.17.2 that I need to figure out what has changed as well and put 1.18.8 there instead.

This is pushed now, and I included our custom modifications to the /config and /metadata folders as well.

**Comment by ElijahLynn** [comment]
> Yay for tests...
> Failure on features/user.feature:4 with:
> 
> > And I should see "Login with PIV or other Smartcard." in the "#edit-simplesamlphp-auth-login-link" element # Drupal\DrupalExtension\Context\MinkContext::assertElementContainsText()
> > ```
> >   Element matching css "#edit-simplesamlphp-auth-login-link" not found. (Behat\Mink\Exception\ElementNotFoundException)
> > ```

There was a new config option that was disabled by default and I had to enable it and re-export it. This is pushed now too and this test passes locally.

File: config/stg/simplesamlphp_auth.settings.yml [hunk 1]
```diff
@@ -32,3 +32,4 @@ role:
   population: ''
   eval_every_time: false
 autoenablesaml: true
+login_link_show: true
```
Comment (indytechcook) [inline config/stg/simplesamlphp_auth.settings.yml:35]: New settings?

Comment (ElijahLynn) [reply config/stg/simplesamlphp_auth.settings.yml:35]: Yeah, and there is an update hook to set the default value but it doesn't check to see if it was set before, at least in our case.

**Comment by ElijahLynn** [comment]
I tested this on DEV and it is working, logging in. 

However, I don't like this pattern of committing simplesamlphp. I think if the folder needs to be outside of docroot, then it should be a post-install-cmd to copy it, and then lay down the config in our devops scripts, ideally. 

This PR is the same as what we have now though, so I don't want to let perfect be the enemy of good.

**Review by swirtSJW — APPROVED** [review]
Approved

**Comment by ElijahLynn** [comment]
> I tested this on DEV and was able to log in.
> 
> However, I don't like this pattern of committing simplesamlphp. I think if the folder needs to be outside of docroot, then it should be a post-install-cmd to copy it, and then lay down the config in our devops scripts, ideally.
> 
> This PR is the same as what we have now though, so I don't want to let perfect be the enemy of good.

K, I finally got this working on DEV. I also took care of this issue and now we are not committing the entire library but just symlinking it and copying the two config/metadata files needed for it to work. This does require https://github.com/department-of-veterans-affairs/devops/pull/7968 to be merged simultaneously.

**Review by indytechcook — APPROVED** [review]

File: composer.json [hunk 2]
```diff
@@ -296,8 +296,7 @@
                 "PHP 7.2 notice - \"Undefined offset: 1\" in JsonSchemaEncoder": "https://www.drupal.org/files/issues/2019-07-08/3063027-5-undefined-offset.patch"
             },
             "drupal/simplesamlphp_auth": {
-                "Don't attempt username sync when user was already matched by username": "https://www.drupal.org/files/issues/2019-08-14/2748731-23.patch",
-                "Bypass password requirement for password change if logged in with sso.": "https://www.drupal.org/files/issues/2018-05-01/simplesamlphp_auth-change-password-access-2968598-2.patch"
```
Comment (swirtSJW) [inline composer.json:300]: These patches are no longer needed?

**Review by swirtSJW — APPROVED** [review]
Nice work getting all those files out of the repo with some symlink magic.

File: composer.json (hunk 2 above)
Comment (ElijahLynn) [reply composer.json:300]: "Bypass password requirement for password change if logged in with sso." is in this new release, but "Don't attempt username sync when user was already matched by username" is not yet committed and is still in this PR, one line below.

**Pull request merged**